optuna/distributions.py

```diff
@@ -278,6 +284,13 @@ def _asdict(self) -> Dict:
 
     @property
     def q(self) -> float:
+        """Discretization step.
+
+        :class:`~optuna.distributions.DiscreteUniformDistribution` is now subtype of
+        :class:`~optuna.distributions.FloatDistribution` and FloatDistribution has ``step``
+        parameter instaead of ``q``. This property is proxy such that users can get a
+        discretization step by ``distribution.q`` for keeping the backward compatibility.
```

Comment:
While it'd make sense in a change log, I'd avoid using "now" here as a part of an API reference. Also, how about making it more compact?
```suggestion
        :class:`~optuna.distributions.DiscreteUniformDistribution` is a subtype of
        :class:`~optuna.distributions.FloatDistribution`.
        This property is a proxy for its ``step`` attribute.
```

Reply:
It really makes sense. Thank you for the suggestion!